src/lib/genesis_ledger_helper/genesis_ledger_helper.ml

```diff
@@ -280,21 +280,31 @@ module Ledger = struct
     let%map () = Tar.create ~root:dirname ~file:tar_path ~directory:"." () in
     tar_path
 
-  let load ~genesis_dir ~logger ~constraint_constants
+  let load ~proof_level ~genesis_dir ~logger ~constraint_constants
       (config : Runtime_config.Ledger.t) =
     Monitor.try_with_join_or_error (fun () ->
         let add_genesis_winner_account accounts =
-          let pk, _ = Coda_state.Consensus_state_hooks.genesis_winner in
-          match accounts with
-          | (_, account) :: _
-            when Public_key.Compressed.equal (Account.public_key account) pk ->
-              accounts
-          | _ ->
-              ( None
-              , Account.create
-                  (Account_id.create pk Token_id.default)
-                  (Currency.Balance.of_int 1000) )
-              :: accounts
+          let add_genesis_winner_account =
+            match config.add_genesis_winner with
+            | Some add_genesis_winner ->
+                add_genesis_winner
+            | None ->
+                Genesis_constants.Proof_level.equal Full proof_level
+          in
+          if add_genesis_winner_account then
+            let pk, _ = Coda_state.Consensus_state_hooks.genesis_winner in
+            match accounts with
+            | (_, account) :: _
+              when Public_key.Compressed.equal (Account.public_key account) pk
+              ->
+                accounts
+            | _ ->
+                ( None
+                , Account.create
+                    (Account_id.create pk Token_id.default)
+                    (Currency.Balance.of_int 1000) )
+                :: accounts
+          else accounts
         in
         let accounts =
```

Comment:
can we name this `accounts_opt`, and the variable on ln 329, `padded_accounts_opt` ?